### DISCUSSION TITLE
Using custom backend plugin (not locize)

### DISCUSSION THREAD

**Discussion by stratienko** [i18next/next-i18next#2078] [Q&A]
Hello! 

I want to use PocketBase as my primary translations source. I wrote my own I18n Backend plugin to communicate with PocketBase and included it in my `next-i18next.config.js` `use` field like this:

```ts
const { PocketBaseI18NextBackend } = require("pocketbase-i18n-backend");

/** @type {import('next-i18next').UserConfig} */
module.exports = {
  backend: {
    collectionName: "translations",
    pocketbaseAdminName: "test@test.test",
    pocketbaseAdminPassword: "test123456",
    pocketbaseUrl: "http://127.0.0.1:8090",
    refetchInterval: 1000 * 60 * 15, // 15 minutes
  },
  i18n: {
    defaultLocale: "en",
    locales: ["en", "es"],
  },
  saveMissing: true,
  serializeConfig: false,
  use: [PocketBaseI18NextBackend],
};
```

Unfortunately, when I call `serverSideTranslations` function, my plugin's `read` method is never executed, because, as I assume, `serverSideTranslations` tries to get translation files from the `public/locales` folder.

My question is: can we somehow tell `serverSideTranslations` to use custom backend's read function instead?

**Comment by adrai** [comment]
it should work like you describe it... is there any error?
Make sure PocketBaseI18NextBackend is not undefined.
Maybe also set debug to true to check the logs.

**Reply by stratienko** [reply]
Already checked, the problem is `serverSideTranslations` creates a `Node.js` client here:

https://github.com/i18next/next-i18next/blob/master/src/serverSideTranslations.ts#L55

It tries to find resources in the `public` folder, never founds them and returns the empty resources object which is passed to the `appWithTranslations` HOC. Since resources are empty, the plugin's `create` function gets executed and pushes fallback translations to the pocketbase but `read` is never actually executed. So, as I understand, there is no way to change the read mechanism for the `serverSideTranslations` and the only way to implement it is to just create my own `pocketbaseTranslations` function that will be executed in the `getServerSideProps`

**Reply by adrai** [reply]
sorry, but I don't fully understand... other backends seems to work... also without having local resources

**Reply by stratienko** [reply]
Yes, but other backends require you to simply delete `serverSideTranslations` function from your `getServerSideProps`. 

For example, locize:

https://github.com/locize/next-i18next-locize/blob/main/pages/index.js#L52

So those resources are loaded on the client side, or am I wrong?

**Reply by adrai** [reply]
If you delete the getServerSideProps function, yes it will only load on client side... but if you keep getServerSideProps it will load also on server side, like possibility 2 here: https://github.com/locize/next-i18next-locize#possibility-2-config-for-locize-live-download-usage
